specs/experimental/contracts/dispute/dispute-game-factory.md

```diff
@@ -0,0 +1,296 @@
+    - [Impact](#impact)
+  - [iDGF-002: Game registry consistency is maintained](#idgf-002-game-registry-consistency-is-maintained)
+    - [Impact](#impact-1)
+- [Function Specification](#function-specification)
+  - [constructor](#constructor)
+  - [initialize](#initialize)
+  - [gameCount](#gamecount)
+  - [games](#games)
+  - [gameAtIndex](#gameatindex)
+  - [create](#create)
+  - [getGameUUID](#getgameuuid)
+  - [findLatestGames](#findlatestgames)
+  - [setImplementation (without args)](#setimplementation-without-args)
+  - [setImplementation (with args)](#setimplementation-with-args)
+  - [setInitBond](#setinitbond)
+
+<!-- END doctoc generated TOC please keep comment here to allow auto update -->
+
+## Overview
+
+The DisputeGameFactory serves as the central registry and factory for creating dispute game contracts in the OP Stack
+fault proof system. It manages game type implementations, tracks all created games, and enforces bond requirements for
+game creation.
+
+## Definitions
+
+### Game UUID
+
+A unique identifier for a dispute game computed as `keccak256(gameType || rootClaim || extraData)`. This identifier
+ensures that each unique combination of game parameters can only result in one game instance.
+
+### Game Implementation
+
+A template contract that implements the IDisputeGame interface. The factory clones this implementation using the
+Clone With Immutable Args (CWIA) pattern to create new game instances efficiently.
+
+### Initialization Bond
+
+The amount of ETH (in wei) required to create a dispute game of a specific game type. This bond is forwarded to the
+created game's initialize function and serves as the initial stake in the dispute.
+
+### Game Clone
+
+A minimal proxy contract created using the CWIA pattern that delegates all calls to a [Game Implementation](#game-implementation)
+while storing immutable game-specific parameters (creator address, root claim, parent hash, extra data) in its bytecode.
+
+## Assumptions
+
+### aDGF-001: Owner operates within governance constraints
+
+The owner of the DisputeGameFactory is trusted to set game implementations and initialization bonds according to
+governance processes. Malicious or incorrect configuration by the owner could compromise the security of the dispute
+system.
+
+#### Mitigations
+
+- Owner is controlled by governance multisig
+- Configuration changes are subject to governance review
+- Monitoring of owner actions
+
+### aDGF-002: Implementation contracts are valid IDisputeGame implementations
+
+Implementation contracts registered via setImplementation are assumed to correctly implement the IDisputeGame interface
+and initialize properly when cloned. Invalid implementations could cause game creation to fail or games to behave
+incorrectly.
+
+#### Mitigations
+
+- Implementation contracts undergo security audits
+- Testing of implementations before registration
+- Gradual rollout of new game types
+
+## Invariants
```

Comment:
devinai the correct invariants are:

- Dispute game factory properly creates dispute games
- Dispute game factory allows anyone to create dispute games
- Dispute game factory games are unique
- Dispute game factory can create games for any game type